src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/GBNode.java

```diff
@@ -0,0 +1,53 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+
+import java.util.Collections;
+import java.util.LinkedList;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.SystemsPosition;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemUtils.getAllSystemsPositions;
+
+public abstract class GBNode {
+    protected boolean isNeighborsSet;
+    protected LinkedList<SystemsPosition> neighbors;
+    protected LinkedList<SystemsPosition> system2MustBeToEnter;
+    protected LinkedList<SystemsPosition> system2MustBeToOut;
+
+
+    public GBNode() {
+        neighbors = new LinkedList<>();
+        isNeighborsSet = false;
+        system2MustBeToEnter = new LinkedList<>(getAllSystemsPositions());
+        system2MustBeToOut = new LinkedList<>(getAllSystemsPositions());
+    }
+
+    public void addNeighbors(SystemsPosition[] neighbors) {
+        if (!isNeighborsSet) {
+            Collections.addAll(this.neighbors, neighbors);
+            isNeighborsSet = true;
+        }
+    }
+
+    public LinkedList<SystemsPosition> getSystem2MustBeToEnter() {
+        return system2MustBeToEnter;
+    }
+
+    public LinkedList<SystemsPosition> getSystem2MustBeToOut() {
+        return system2MustBeToOut;
+    }
+
+    public void setSystem2MustBeToEnter(SystemsPosition[] system2MustBeToEnter) {
```

Comment:
Setting is always for something, you can just write 'setMustFroEnter' or something like that

Reply:
Same for the function below